### PR TITLE
fix(python,rust): sql window functions

### LINKED ISSUE #7436: Polars SQL: Window function disregards the partition argument
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

The SQL function seems to disregard the partition argument. 

### Reproducible example

```python
import polars as pl

df = pl.read_csv(
"https://gist.githubusercontent.com/ritchie46/cac6b337ea52281aa23c049250a4ff03/raw/89a957ff3919d90e6ef2d34235e6bf22304f3366/pokemon.csv"
)

sql = pl.SQLContext()
sql.register('pokemon', pokemon)

sql.query("""
SELECT 
    "Type 1" AS type1,
    AVG(Attack) OVER (PARTITION BY "Type 1") AS avg_attack_by_type
FROM pokemon
LIMIT 5
""")

shape: (5, 2)
┌───────┬────────────────────┐
│ type1 ┆ avg_attack_by_type │
│ ---   ┆ ---                │
│ str   ┆ f64                │
╞═══════╪════════════════════╡
│ Grass ┆ 75.349693          │
│ Grass ┆ 75.349693          │
│ Grass ┆ 75.349693          │
│ Grass ┆ 75.349693          │
│ Fire  ┆ 75.349693          │
└───────┴────────────────────┘

# Groupby works well
sql.query("""
SELECT 
    "Type 1",
    AVG(Attack) AS avg_attack_by_type
FROM pokemon
WHERE "Type 1" = 'Grass' OR "Type 1" = 'Fire'
GROUP BY "Type 1"
""")

shape: (2, 2)
┌────────┬────────────────────┐
│ Type 1 ┆ avg_attack_by_type │
│ ---    ┆ ---                │
│ str    ┆ f64                │
╞════════╪════════════════════╡
│ Grass  ┆ 72.923077          │
│ Fire   ┆ 88.642857          │
└────────┴────────────────────┘
```

### Expected behavior

I expect the value for Grass to be 72.92 and the value for Fire to be 88.64 in the window function

### Installed versions

<details>

```
---Version info---
Polars: 0.16.11
Index type: UInt32
Platform: Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.31
Python: 3.11.1 (main, Dec  7 2022, 01:11:44) [GCC 9.4.0]
---Optional dependencies---
pyarrow: 11.0.0
pandas: 1.5.1
numpy: 1.24.0
fsspec: <not installed>
connectorx: <not installed>
xlsx2csv: 0.8.1
deltalake: <not installed>
matplotlib: 3.6.2
```

</details>

### LINKED ISSUE #7437: Polars SQL: groupby returns a list when the groupby column has an alias
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

When an alias is assigned to the groupby column, a list is return instead of only the distinct values

### Reproducible example

```python
import polars as pl

df = pl.read_csv(
"https://gist.githubusercontent.com/ritchie46/cac6b337ea52281aa23c049250a4ff03/raw/89a957ff3919d90e6ef2d34235e6bf22304f3366/pokemon.csv"
)

sql = pl.SQLContext()
sql.register('pokemon', pokemon)

# Working OK
sql.query("""
SELECT 
    "Type 1"
FROM pokemon
GROUP BY "Type 1"
""")

shape: (15, 1)
┌──────────┐
│ Type 1   │
│ ---      │
│ str      │
╞══════════╡
│ Grass    │
│ Ghost    │
│ Poison   │
│ Water    │
│ ...      │
│ Electric │
│ Rock     │
│ Dragon   │
│ Fighting │
└──────────┘

# Returning a list -> not ok
sql.query("""
SELECT 
    "Type 1" AS type_1
FROM pokemon
GROUP BY "Type 1"
""")

shape: (15, 1)
┌─────────────────────────────────────┐
│ type_1                              │
│ ---                                 │
│ list[str]                           │
╞═════════════════════════════════════╡
│ ["Dragon", "Dragon", "Dragon"]      │
│ ["Grass", "Grass", ... "Grass"]     │
│ ["Ice", "Ice"]                      │
│ ["Ground", "Ground", ... "Ground... │
│ ...                                 │
│ ["Normal", "Normal", ... "Normal... │
│ ["Psychic", "Psychic", ... "Psyc... │
│ ["Rock", "Rock", ... "Rock"]        │
│ ["Bug", "Bug", ... "Bug"]           │
└─────────────────────────────────────┘
```

### Expected behavior

The groupby column should show the distinct values only, and not a list

### Installed versions

<details>

```
---Version info---
Polars: 0.16.11
Index type: UInt32
Platform: Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.31
Python: 3.11.1 (main, Dec  7 2022, 01:11:44) [GCC 9.4.0]
---Optional dependencies---
pyarrow: 11.0.0
pandas: 1.5.1
numpy: 1.24.0
fsspec: <not installed>
connectorx: <not installed>
xlsx2csv: 0.8.1
deltalake: <not installed>
matplotlib: 3.6.2
```

</details>

### PR DESCRIPTION
**Pull request by universalmind303**
closes #7436, 
closes #7437

The order of operations was wrong, we were applying the function after the window function 

`avg(a) over (partition by b)`
would result in the polars expression `col("a").over([col("b")]).mean()` instead of `col("a").mean().over([col("b")])`

@ritchie46 i also added the pokemon csv to `examples/datasets/csv`. If you would prefer this to not be included, I can create an alternative test for this.

### CODE REVIEW

File: polars/polars-lazy/polars-plan/src/dsl/meta.rs [hunk 1]
```diff
@@ -22,6 +22,12 @@ impl MetaNameSpace {
     pub fn root_names(&self) -> Vec<Arc<str>> {
         expr_to_leaf_column_names(&self.0)
     }
+    /// A projection that only takes a column or a column + alias.
+    pub fn is_simple_projection(&self) -> bool {
+        let mut arena = Arena::with_capacity(8);
+        let node = to_aexpr(self.0.clone(), &mut arena);
```
Comment (universalmind303) [inline polars/polars-lazy/polars-plan/src/dsl/meta.rs:28]: @ritchie46 is there any way to do this without cloning the expr?

**Comment by ritchie46** [comment]
> @ritchie46 i also added the pokemon csv to examples/datasets/csv. If you would prefer this to not be included, I can create an alternative test for this.

Yes, I think we have enough test files already. Could you rewrite the test?

And thanks for the PR. :)

**Comment by universalmind303** [comment]
@ritchie46 this should be good to go now.

File: polars/polars-lazy/polars-plan/src/dsl/meta.rs (hunk 1 above)
Comment (ritchie46) [reply polars/polars-lazy/polars-plan/src/dsl/meta.rs:28]: Yes, implementing the `is_simple_projection` for the `Expr` type. Maybe a good follow up PR?

**Pull request merged**